### PR TITLE
Add new genre Detour

### PR DESCRIPTION
**Pull request by boboquack**
Some things still need Japanese translation: the genre name "Detour" and the specific error message "The room has the wrong number of turns", but apart from that this has been briefly tested locally and appears to work.

### CODE REVIEW

**Comment by boboquack** [comment]
Apparently there are some code style issues in `src/variety/country.js` according to `prettier --check '{src,src-ui,test}/**/*.{js,css}'`. I don't know how to fix that but I don't think it affects performance??

**Comment by boboquack** [comment]
Well, I used prettier, so idk what happened there

**Comment by boboquack** [comment]
Apparently it wants trailing commas at the ends of all list items (including the last items) but that breaks the file

**Comment by robx** [comment]
I don't see any trailing comma issues with prettier. Checking out your branch locally, `make format; make` works (and fixes mostly spacing around function declarations).

**Comment by boboquack** [comment]
For me, that complains that it can't find any files to prettify:
```
> pzpr@0.12.0 format C:\Users\user\Documents\GitHub\pzprjs
> prettier --write '{src,src-ui,test}/**/*.{js,css}'

[error] No matching files. Patterns tried: '{src,src-ui,test}/**/*.{js,css}' !**/node_modules/** !./node_modules/** !**/.{git,svn,hg}/** !./.{git,svn,hg}/**
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! pzpr@0.12.0 format: `prettier --write '{src,src-ui,test}/**/*.{js,css}'`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the pzpr@0.12.0 format script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2020-04-21T22_52_09_386Z-debug.log
make: *** [Makefile:16: format] Error 2
```

**Comment by boboquack** [comment]
Apparently it's an issue with quotes, I already had to edit parts of `package.json` to work with windows.
The relevant lines, for future users:
```
    "build": "eslint --cache --quiet src src-ui && git-hash.sh && grunt default",
    "format": "prettier --write \"{src,src-ui,test}/**/*.{js,css}\"",
    "check-format": "prettier --check \"{src,src-ui,test}/**/*.{js,css}\"",
```

**Comment by robx** [comment]
a separate PR to fix things for windows would be great then

**Comment by boboquack** [comment]
Hmm, I'll have to learn how makefiles and github branches work in more detail then... i'll try to work that out soon.

**Comment by robx** [comment]
> Hmm, I'll have to learn how makefiles and github branches work in more detail then... i'll try to work that out soon.

Oh in that case, happy to (try to) fix it myself, no worries.

**Comment by robx** [comment]
I pushed a fix for the quote change to master. Not tested on windows, but still works for me at least.

File: src/variety/country.js [hunk 1]
```diff
@@ -7,1062 +7,1129 @@
...
+			],
+			blLineLong: [
+				"枠内を連続して通るマス数が、数字よりも大きい線があります。",
+				"A line in a room is longer than the number."
+			]
+		},
+		"FailCode@detour": {
+			blWrongTurns: [
+				"The room has the wrong number of turns.",
+				"The room has the wrong number of turns."
```
Comment (robx) [inline src/variety/country.js:1131]: I think for consistency in style this should be "A room has the wrong number of turns."

**Comment by robx** [comment]
In other things, I'd really like to have the sub marks available (the cell circle/cross marks), to be able to mark which cells are turns and which aren't. (Something less visually heavy than the X *might* be nice, but I wouldn't worry about that now.)

Looks great apart from that! (Well the diff looks awful but that's `prettier`'s fault :/... Might change the start of the file to read

```
(function(classbase) {
        var pidlist = [
                "country",
                "moonsun",
                "onsen",
                "doubleback",
                "maxi",
                "simpleloop",
                "detour"
        ];
        if (typeof module === "object" && module.exports) {
                module.exports = [pidlist, classbase];
        } else {
                pzpr.classmgr.makeCustom(pidlist, classbase);
        }
})({
```

but we shouldn't really have to work around the formatter.)

File: src/variety/country.js (hunk 1 above)
Comment (boboquack) [reply src/variety/country.js]: Cool, changed

**Comment by robx** [comment]
merged by hand, thanks!

**Pull request closed without merging**